web/components/SchedulePage.jsx

```diff
@@ -67,10 +56,130 @@ class SchedulePage extends React.Component {
         this.props.setSelectedDay(days[0])
     }
 
+    componentWillReceiveProps(nextProps) {
+        const { currentTeamMember, editModeSchedule } = this.props;
+        // This is to make a different array for editing purpose, completed with empty set blocks
+        // This only take effect if change the currentTeamMember
+        if (editModeSchedule && currentTeamMember !== nextProps.currentTeamMember) {
+            this.makeSetBlocksForEdit(nextProps.currentTeamMember);
+        }
+    }
+
+    makeSetBlocksForEdit = (currentTeamMember) => {
+        const { daysOfWeek } = this.state
+        const setBlocksByDate = _.groupBy(currentTeamMember.weeklySetblocks, 'date')
+        daysOfWeek.map((day) => {
+            day = moment(day).format('YYYY-MM-DD')
+            setBlocksByDate[day] = this.completeWithEmptySetBlocks(setBlocksByDate[day])
+        })
+        this.setState({
+            setBlocksForEdit: setBlocksByDate
+        })
+    }
+
+    completeWithEmptySetBlocks(setBlocks) {
+        const defaultSetBlocks = [
```

Comment:
Let's move this into the /web/constants/index.js and import it outside of the component as `DEFAULT_SETBLOCKS` since it's a pretty common and I'm sure it will be reused elsewhere